napalm-ruckus-fastiron/ruckus_fastiron.py

```diff
@@ -0,0 +1,1405 @@
+        elif retrieve == 'candidate':
+            config_list.append('')
+        elif retrieve == 'all':
+            config_list.append('show running-config')
+            config_list.append(None)
+            config_list.append('show config')
+
+        for cmd in config_list:
+
+            if cmd is None:
+                config_dic.update({'candidate': {}})
+                continue
+
+            output = self.device.send_command(cmd)
+            n_line = FastIronDriver.__creates_list_of_nlines(output)
+
+            if cmd == 'show running-config':
+                config_dic.update({'running': n_line})
+            elif cmd == '':
+                config_dic.update({'candidate': n_line})
+            else:
+                config_dic.update({'startup': n_line})
+
+        return config_dic
+
+    def get_network_instances(self, name=''):
+        """
+        Return a dictionary of network instances (VRFs) configured, including default/global
+
+        Args:
+            name(string) - Name of the network instance to return, default is all.
+
+        Returns:
+            A dictionary of network instances in OC format:
+            * name (dict)
+              * name (unicode)
+              * type (unicode)
+              * state (dict)
+                * route_distinguisher (unicode)
+              * interfaces (dict)
+                * interface (dict)
+                  * interface name: (dict)
+
+        Example:
+        {
+            u'MGMT': {
+                u'name': u'MGMT',
+                u'type': u'L3VRF',
+                u'state': {
+                    u'route_distinguisher': u'123:456',
+                },
+                u'interfaces': {
+                    u'interface': {
+                        u'Management1': {}
+                    }
+                }
+            }
+            u'default': {
+                u'name': u'default',
+                u'type': u'DEFAULT_INSTANCE',
+                u'state': {
+                    u'route_distinguisher': None,
+                },
+                u'interfaces: {
+                    u'interface': {
+                        u'Ethernet1': {}
+                        u'Ethernet2': {}
+                        u'Ethernet3': {}
+                        u'Ethernet4': {}
+                    }
+                }
+            }
+        }
+        """
+        vrf_dict = dict()                                           # Dictionary that will append
+        vrf_interface = dict()
+        check = self.device.send_command('show version')
+
+        if check.__contains__('7150'):                              # ICX7150 does not support VRF
+            return {}
+
+        if check.__contains__('SPR'):                               # check if router version
+            vrf_type = 'L3VRF'                                      # adds vrf of type router
+        else:
+            vrf_type = 'L2VRF'                                      # adds vrf of type switch

```

Comment:
When IETF/openconfig is specifying L3VRF/L2VRF, they're specifically referring to L3VPNs and L2VPNs (EVPN, psuedowire, etc), not the model/type of device containing the VRF.  I imagine all the VRFs on this device would be the L3VRF type.

Reply:
Thanks for the tip. Ruckus ICX is unique in that not all models support L3/L2VRF or any device running on switch image. I have added the following code that will return an empty dictionary if VRF is not supported. Currently working on cases for VRF configured 

`        if any(x in check for x in ["7150", "SPS"]):                # ICX7150 does not support VRF
            return {}                                               # neither does switch image

        if "7250" in check:
            cur_version = FastIronDriver.__retrieve_all_locations(check, 'Version', 1)

            if cur_version.pop() > "8.0.50":
                pass
            else:
                return {}`